addons/mail/models/discuss/discuss_channel.py

```diff
@@ -485,6 +485,9 @@ def get_vals(channel):
         old_vals = {channel: get_vals(channel) for channel in self}
         result = super().write(vals)
         for channel in self:
+            if (name := vals.get("name")) and channel.channel_type in ("channel", "group"):
```

Comment:
We might want to check name != channel.name (before write) to avoid unnecessary notification if writing name to existing value.